src/rv32_3stage/arbiter.scala

```diff
@@ -25,56 +24,58 @@ class SodorMemArbiter(implicit val conf: SodorConfiguration) extends Module
    }
 
    //***************************
-   // Figure out who gets to go
-   val req_fire_dmem = Bool()
-   val req_fire_imem = Bool()
-
-   // default
-   req_fire_dmem := Bool(false) 
-   req_fire_imem := Bool(false) 
-
-   when (io.dmem.req.valid)
+   val i1reg = Reg(Bits(width=conf.xprlen))
+   val d1reg = Reg(Bits(width=conf.xprlen))
+   val nextdmem = Reg(init=Bool(true))
+   io.dmem.req.ready := Bool(true)
+   val d_or_i = new Bool()
```

Comment:
What does the `d_or_i` variable do? What does it mean when it is true (`dmem & nextdmem`?)? I'm finding the name confusing. Is something like `d_wins` or `d_fire` more clear?